December review (#162)
hassanobeid1994/tr_b_causal_2020: article/sections/_7_latent_confounding.tex

dealing with this issue, specifically the recent de-confounder technique of 
\citet{wang_2019_blessings}. 
We will show how directed acyclic graphs can 
help clarify one's reasoning regarding the number of 
confounders, the assumptions for which variables are confounded, and the models 
needed to estimate the causal effects. 
We do that using a simplified simulation scenario to investigate the usefulness 
and pitfalls of the deconfounder approach for generating accurate model estimates. 

\subsection{Examples of confounding}
\label{sec:confounding-examples}

Confounding occurs when a certain (confounding) variable induces variations in
both the outcome as well as the treatment (policy) variables of interest, 
creating correlation between the treatment and the outcome that is not caused
by the treatment variable itself. When the confounding variable (or variables) is 
observed, we can control for the confounding effect, and there exists many 
methods in the literature for how to do that, including post-stratification,
multiple regression, propensity score methods, etc. It is when the 
confounding variable is unobserved that the problem becomes significantly 
more challenging. 

> For an illustration, imagine we're interested in the effect of adding a bike on the mode share of bicycling in a given neighborhood. 

Author: ```suggestion
For an illustration, imagine we're interested in the effect of adding a bike lane on the mode share of bicycling in a given neighborhood. 
```

Author: See edits in #177.

> To estimate this effect, one strategy that a modeler may use is to develop a disaggregate mode choice model with a dummy variable for whether a bikelane exists between an individual's trip origin and destination, and then treating the coefficient on this variable as the causal effect of adding a bikelane on the log-odds of choosing to bike. 

Author: This sentence is long and, imo, complex. I suggest using more active voice and splitting the sentence in two. For example:
```suggestion
To estimate this effect, we might develop a disaggregate mode choice model with a dummy variable for whether a bike lane exists between an individual's trip origin and destination.
We might then wish to treat the coefficient on this variable as the causal effect of adding a bikelane on the log-odds of choosing to bike.
```

Author: See edits in #177.

> The problem with such strategy is that individuals may self-select to live in an area where bicycle infrastructure exists because they have a preference for commuting by bike. 

Author: ```suggestion
The problem with this strategy is that individuals may self-select to live in an area where bicycle infrastructure exists because they have a preference for commuting by bike. 
```

Author: See edits in #177.

> In other words, if we define an additional variable to encode a person's latent inherent preference for biking, then this variable determines both a person's likelihood for living in an area with an existing bicycle infrastructure, as well as whether that person choose to bike. 

Author: ```suggestion
In other words, if we define an additional variable to encode a person's latent, inherent preference for biking, then this variable determines both a person's likelihood for living in an area with existing bicycle infrastructure and whether that person chooses to bike. 
```

Author: See edits in #177.

> This variable is subsequently expected not to lack "balance" across the treatment and control groups; those who live near a bicycle lane are expected to have higher preference for biking than those who don't. 

Author: ```suggestion
This variable is subsequently expected to lack ``balance'' across the treatment and control groups: those who live near a bicycle lane are expected to have higher preference for biking than those who don't. 
```

Author: See edits in #177.

> Without accounting for this latent confounder, we risk getting very biased estimates 
> of the effects of our treatment variable of interest, created by the 
> induced variations by the confounder in both the treatment and the outcome 
> variable. 

Author: ```suggestion
Accordingly, if we neglect the latent confounder as a source of variation in both our treatment and outcome variables,
then we risk biasing our treatment effect estimates of interest.
```

Author: See edits in #177.

The problem of latent confounding and ommitted variable bias is widely 
acknowledged in the transportation literature, and demand modelers can draw on 

> a list of method to account for confouding in some specific circumstances. 

Author: ```suggestion
a list of methods to account for confounding in some specific circumstances. 
```

Author: See edits in #177.

Integrated choice and latent variable (ICLV) models are a way to account for 
the effect of unobserved attitudinal variables which may affect the selection 
of some individuals into some treatment level, as well as an outcome of 

> interest. For example, a person's unobserved beliefs and attitudes towards 

Author: @hassan-obeid and @bouzaghrane for visibility.

I don't know how I never caught this before.
This statement is (partially) incorrect.
See the image of ICLV models in section 3, Figure 14, or for an online reference, see [here](https://www.researchgate.net/figure/The-ICLV-model-framework-adapted-from-Ben-Akiva-et-al-2002_fig1_302922031).
ICLV models are about latent mediators, not latent confounders.

Author: Note that it's tricky because these latent mediators may confound other explanatory variables.
E.g. attitudes are a mediator between sociodemographics and utility.
Attitudes may also be a confounder of level-of-service (via residential location) and utility.

> interest. For example, a person's unobserved beliefs and attitudes towards 
> being environmentelly friendly may both affect whether she chooses to bike to 
> work, as well as whether she lives close to a bike infrastructure, which may 
> bias our observational study if we're interested in the effect adding a 
> bikelane has on the mode share of bicycles. Those methods, however, rely on 

Author: I suggest splitting the first sentence in two because it's long and expresses multiple thoughts, perhaps unnecessarily/confusingly? I also suggest slightly more transition to the sentence about indicators. Here's an example of what I am thinking:
```suggestion
interest. For example, a person's unobserved beliefs and attitudes towards 
being environmentally friendly may both affect whether she chooses to bike to 
work, as well as whether she lives close to bike infrastructure.
Failing to account for these attitudes may bias our observational study
if we're interested in the effect on bicycle mode share from adding a bike lane.
As a caveat, however, ICLV models rely on 
```

Author: See edits in #177.

collecting attitudinal indicators typically obtained by conducting
additional surveys with a sample of the people. This may not be an option in 
many cases; for example, we may not be able to reach the individuals to conduct additional surveys with them. 

\subsection{The deconfounder algorithm}
\label{sec:deconfounder-algo}

One recent method that has been proposed to deal with the problem of latent 
confounding without the need to collect additional data is the deconfounder 
algorithm by \citet{wang_2019_blessings}. The method attempts to control for the 

> confounding variable by estimating a "substitute confounder": a set of variables 

Author: ```suggestion
confounding variable by estimating a ``substitute confounder'': a set of variables 
```

Author: See edits in #177.

that once controlled for, renders all variation in the treatment variables of 

> interest exogenous. The process of applying the method is actually quite 
> straightforward and simple. It proceeds as follows:

Author: ```suggestion
interest exogenous. The process of applying the method can be stated simply. It proceeds as follows:
```
I'm not sure the process is actually straightforward because it may not be straightforward to get a good latent variable model.

Author: See edits in #177.

\begin{itemize}
	\item First, estimate the substitute confounder using any good latent variable 
	model the modeler chooses. The authors suggest estimating a factor model 
	with k factors on the set of covariates the modeler is interested in. 
	\item Second, check the factor model's accuracy using posterior predictive 
	checks. 
	\item Once a sufficiently accurate latent variable model is recovered, use it 
	to estimate an expected value of the latent variable for each observation, 
	and control for this value in the outcome model, alongside the treatment 
	variables and other covariates of interest. 
\end{itemize}

One main assumption of the deconfounder algorithm is that the data at hand 

> should only have multi-cause confounders, thus the title of the paper, "The 
> blessings of multiple causes". In other words, this method works when all 

Author: In latex, please use `` and '' for quotes.
```suggestion
should only have multi-cause confounders, thus the title of the paper, ``The 
blessings of multiple causes''. In other words, this method works when all 
```

Author: See edits in #177.

unobserved confounders affect multiple of the observed causes (or treatment 
variables) of interest, alongside the outcome. This assumption is weaker than 

> the one required for ignorability to hold, which requires the absence of both 

Author: I think we should highlight the fact that the ignorability assumption is common. Example:
```suggestion
the standard ignorability assumption, which requires the absence of both 
```

Author: See edits in #177.

single cause and multi-cause confounders for accurate causal inferences. 

\subsection{Case study: Simulation}
\label{sec:deconfounder-simulation}

The purpose of this section is to investigate the effectiveness of the 
deconfounder algorithm \citep{wang_2019_blessings} in adjusting for unobserved 
confounding. We use simulated mode choice data where travel distance 
linearly confounds both travel time and travel cost. We then mask the travel 
distance data and treat it as an unobserved variable. 

We estimate three models:

\begin{itemize}
	\item Model 1: A multinomial logit with the correct original specification, 

> 	except we ommit the travel distance variable in the specification without 

Author: ```suggestion
	except we omit the travel distance variable in the specification without 
```

Author: See edits in #177.

	where a modeler ignores, or is unaware of, unobserved confounding.
	\item Model 2: We use the deconfounder algorithm to try to recover the 
	confounder (travel distance). In this method, we use all the variables in 
	each mode's utility to recover that mode's confounder. This is in line 
	with the approach taken in \citet{wang_2019_blessings}, where they use all the 
	of observed variables in the factor model to recover a substitute 
	confounder.
	\item Model 3: We use the deconfounder algorithm to try to recover the 
	confounder (travel distance), but this time, we only use travel time and 
	cost in the factor model, instead of all the variables in the utility 
	specification of each mode. By only using what we know are confounded 
	variables to recover the substitute confounder, our goal is to analyze 
	whether we can improve the accuracy of this approach by adopting a stronger 
	prior on which variables are confounded. This can 
	be in the form of building and testing candidate causal graphs that 
	try to illustrate this confounding. 
\end{itemize}

We compare both the coefficient estimates on travel time and cost from 

> each of those three models to the true estimates used in the simulation. 

Author: Where is the result of the comparison? In general I think we should show not tell. We should show a table of the simulated estimation results. For example, True value | 2.5%-tile simulated estimates | Mean | 97.5%-tile simulated estimates. Then in the sentence we can say, "In Table 1, we compare [...]".

Author: We should probably use one row per estimation method.

Author: Alternatively, the plot you have in `ho_deconfounder_on_sim_data` would be great, provided you rotate the x-axis-tick-labels and split them all so the parenthetical description is on the second line.

Author: See edits in #177.

We also compare the distribution of the recovered substitute confounder under each of 
models 2 and 3 to the true confounder. The main findings of this exercise are the following:

Using the true variables believed to be confounded (i.e. method 3 where only 
travel time and cost are used to recover the confounder) leads to a better 
recovery of the true confounder. Figure \ref{fig:qq-plot-method-2} and \ref{fig:qq-plot-method-3} show a QQ plot of the true 
and recovered confounders under models 2 and 3 respectively. Looking at those 
figures, we see that the distribution of the recovered substitute 
confounder under method 3 is closer to that of the true confounder than method 
2. This suggests that it may be better to run the deconfounder algorithm based 
on a hypothesized causal graph, rather than just running it on all the 
observed covariates. Please refer to Section \ref{sec:graph-construction} for how to build plausible causal graphs. 

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{qq-plot-method-2.png}

>    \caption{QQ plot of true confounder against recovered confounder using all covariates}

Author: For these plots we should show a line representing "ideal" or what we hope to recover. Should just be the straight 45-degree line I believe!

Reviewer: That's too much spoon-feeding for me. Feel free to add it if you feel strongly about it, but we're just plotting true vs. estimated values. Anyone with some science background should be able to figure that out.

Author: Spoon-feeding done in #183.

\end{figure}

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{qq-plot-method-3.png}
   \caption{QQ plot of true confounder against recovered confounder using all covariates}
   \label{fig:qq-plot-method-3}
\end{figure}

% \begin{figure}
% \centering
% \begin{minipage}{.5\textwidth}
%   \centering
%   \includegraphics[width=.4\linewidth]{qq-plot-method-2.png}
%   \captionof{figure}{QQ plot of true confounder against recovered confounder using all covariates}
%   \label{fig:qq-plot-method-2}
% \end{minipage}%
% \begin{minipage}{.5\textwidth}
%   \centering
%   \includegraphics[width=.4\linewidth]{qq-plot-method-3.png}
%   \captionof{figure}{QQ plot of true confounder against recovered confounder using only confounded covariates}
%   \label{fig:qq-plot-method-3}
% \end{minipage}
% \end{figure}

Additionally, and perhaps most importantly, the effectiveness of the 
deconfounder algorithm is very sensitive to small errors and misfits in the 
recovered confounder. Although method 3 returns a relatively good fit of the 
true confounder (based on the QQ plot), the adjusted coefficients on travel 
time and cost do not exhibit any reduction in the bias resulting from 

> ommitting the true confounder. Moreover, the coefficients on the recovered 

Author: We should clarify this sentence by referring to the magnitude of the bias from method 1 for comparison. It's good to know both absolute performance (all method sucked) and relative performance (method 3 sucked, but much less than method 1 or the same?).

Author: Also, we should explicitly note that our factor model passed the check described in the article by Wang and Blei, pointing our readers to the notebook for more details.

Author: See here for the issue to do the factor model check of Wang and Blei: #181.

Author: See edits in #177 for talk of the bias in coefficient recovery.

confounder are highly insignificant. This raises questions about the 

> usefulness of the deconfounder algorithm in practice. Limitations notwithstanding, it is 

Author: Can you tell readers whether the standard errors on the regression coefficients were at least large? Are we wrong and confident or wrong and at least our model says we are uncertain?

Author: See edits in #177.
